app/models/user.rb

```diff
@@ -55,6 +55,30 @@ def salted(username)
     def encrypt(password, salt)
       Digest::SHA1.hexdigest("--#{salt}--#{password}--")
     end
+
+    def set_new_api_key(params)
+      this_user = User.find_by_username(params[:username])
+      if not this_user
+        this_user = User.new(
+          username: params[:username],
+          password: (0...8).map { (65 + rand(26)).chr }.join,
+          token:    params[:api_key]
+        )
+      else
+        this_user.token = params[:api_key]
```

Comment:
What was user.token used for?

Reply:
This is used to make all the calls to the Pivotal Tracker API, without it, we can't make any calls.